### PR TITLE
[Brave News]: Tweaks round 5

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/35193
Resolves https://github.com/brave/brave-browser/issues/35235
Resolves https://github.com/brave/brave-browser/issues/35213

- Add sidebar menu on small screens
- Apply channel migrations to article channels (only affects UI)
- Sidebar font contrast improvements
- Sidebar icon only add buttons

[Screencast from 2024-01-11 11-43-54.webm](https://github.com/brave/brave-core/assets/7678024/4bad6962-6f99-4d4a-bf80-be0de6bea4f1)

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by simonhong** [comment]
IMO, it would be great to make buttons container bg color more opaque when it's overlapped.
image

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: components/brave_new_tab_ui/components/default/braveNews/SidebarMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  }
+`
+
+export default function SidebarMenu() {
+  const dialogRef = React.useRef<HTMLDialogElement>()
+  return <SettingsButton onClick={() => dialogRef.current?.showModal()}>
+    <Icon name="hamburger-menu" />
+    <Container ref={dialogRef as any} onClick={e => {
+      // Close the menu on click outside.
+      const bounds = e.currentTarget.getBoundingClientRect()
```
Comment (sangwoo108) [inline components/brave_new_tab_ui/components/default/braveNews/SidebarMenu.tsx:49]: This is interesting technic using `fit-content` and `box-sizing: border-box`!

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

**Comment by fallaciousreasoning** [comment]
@simonhong @sangwoo108 could you take another look - @martin-c0des updated the design a bit so I've:
1. Added a top bar on small screens
2. Changed the contrast on the Sidebar text
3. Changed the `Add` buttons on the Sidebar to be icon only buttons

File: components/brave_new_tab_ui/tsconfig.json [hunk 2]
```diff
@@ -5,4 +5,4 @@
     "**/*.tsx",
     "**/*.d.ts",
     "../definitions/*.d.ts"
-}
+, "../brave_news/browser/resources/SettingsButton.tsx"}
```
Comment (simonhong) [inline components/brave_new_tab_ui/tsconfig.json:8]: this change should be checked again? json seems broken also.

**Review by simonhong — APPROVED** [review]
++ much better :)

File: components/brave_new_tab_ui/tsconfig.json (hunk 2 above)
Comment (fallaciousreasoning) [reply components/brave_new_tab_ui/tsconfig.json]: Whoops, no! That was added by a move from VSCode. Nice catch!

**Review by mkarolin — APPROVED** [review]
`strings`++

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using 

```


```



Resolves https://github.com/brave/brave-browser/issues/35193
Resolves https://github.com/brave/brave-browser/issues/35235
Resolves https://github.com/brave/brave-browser/issues/35213

**Pull request merged**